dashboard/src/t5gweb/libtelco5g.py

```diff
@@ -191,6 +191,49 @@ def get_case_number(link, pfilter='cases'):
                 return parsed_url.path.split('/')[3]
     return ''
 
+def add_watcher_to_case(cfg, case, username, token):
+    """
+    Add a new watcher to a Red Hat support case.
+
+    :param cfg: The configuration dictionary.
+    :param case: The case number.
+    :param username: The SSO username of the user to add as a watcher.
+    :param token: The Red Hat API token.
+
+    :return: True if the user was successfully added as a watcher, False otherwise.
+    """
+
+    # Get the details of the cases from Redis
+    cases_details = redis_get('details')
+
+    # Get details of the case
+    case_detail = cases_details.get(case)
+
+    # Check if the watcher is already added
+    current_watchers = {user.get('ssoUsername', '') for user in case_detail.get('notified_users', [])}
```

Comment:
~~~
Message: 'Task %(name)s[%(id)s] %(description)s: %(exc)s'
Arguments: {'hostname': 'celery@taskmgr-59d4db95c8-n66lf', 'id': '7bfc0df9-0181-4c69-8c8b-70bec6732d83', 'name': 't5gweb.taskmgr.portal_jira_sync', 'exc': 'AttributeError("\'NoneType\' object has no attribute \'get\'")', 'traceback': 'Traceback (most recent call last):\n  File "/usr/local/lib/python3.11/site-packages/celery/app/trace.py", line 451, in trace_task\n    R = retval = fun(*args, **kwargs)\n                 ^^^^^^^^^^^^^^^^^^^^\n  File "/usr/local/lib/python3.11/site-packages/celery/app/trace.py", line 734, in __protected_call__\n    return self.run(*args, **kwargs)\n           ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File "/srv/t5gweb/taskmgr.py", line 139, in portal_jira_sync\n    message_content, new_cards = libtelco5g.create_cards(cfg, new_cases, action=\'create\')\n                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File "/srv/t5gweb/libtelco5g.py", line 278, in create_cards\n    add_watcher_to_case(cfg, case, assignee[\'jira_user\'], token)\n  File "/srv/t5gweb/libtelco5g.py", line 213, in add_watcher_to_case\n    current_watchers = {user.get(\'ssoUsername\', \'\') for user in case_detail.get(\'notified_users\', [])}\n                                                                ^^^^^^^^^^^^^^^\nAttributeError: \'NoneType\' object has no attribute \'get\'\n', 'args': '[]', 'kwargs': '{}', 'description': 'raised unexpected', 'internal': False}
~~~

Reply:
probably need the case cache to refresh first

Reply:
It seems like the issue is related to failing to get the case details (case_detail = cases_details.get(case)) from redis, so the notified_users key to not exist. I will remove that part and simply post the watcher instead.

Reply:
Obtaining case details from redis has been removed. the task for obtaining case details runs twice a day while the card and case synchronization runs hourly or more frequently. The 'notified_users' key may not exist yet if case details were updated outside of the synchronization cycle.